src/Reflow/Parser.hs

```diff
@@ -31,7 +31,10 @@ header = do
     return $ Header $ n <> v
 
 headerName :: Parser Text
-headerName = pack <$> (many1 $ noneOf ":\n")
+headerName = do
+    n <- manyTill (noneOf "\n") (lookAhead $ string ":")
```

Comment:
so many questions!

@pbrisbin I _do_ want the `:` here. I'm not actually caring about splitting the header into name/value pairs, I just care about recognizing them and then reconstructing the line. I do like the idea of using the specific character set though, I'll make that change.

@gabebw `noneOf "\n:"` will consume input till it sees either `:` _or_ `\n`. This isn't actually what we want though, since we want to make sure we're parsing strings that look like `"foo: bar\n"` and _not_ `"f\noo: bar"`.

@jferris My understanding is that `lookAhead` is needed because without it, we'll consume and then throw away the trailing ":". I suppose I could re-add it manually though?